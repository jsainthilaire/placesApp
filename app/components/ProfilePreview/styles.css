.containerProfilePreview {
    align-items: center;
    display: flex;
    flex-direction: column;
    text-align: center;
}

.profileImage {
    border-radius: 100%;
    width: 200px;
}
