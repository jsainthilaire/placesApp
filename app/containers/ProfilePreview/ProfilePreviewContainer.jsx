import React, { Component } from 'react'
import { connect } from 'react-redux'
import { ProfilePreview } from 'components'

class ProfilePreviewContainer extends Component {
  render() {
    return (
      <ProfilePreview {...this.props} />
    )
  }
}

const mapStateToProps = ({ auth }) => ({
  name: auth.displayName,
  email: auth.email,
  photoURL: auth.photoURL,
})

export default connect(mapStateToProps)(ProfilePreviewContainer)
