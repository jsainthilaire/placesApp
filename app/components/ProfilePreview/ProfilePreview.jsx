import React from 'react'
import PropTypes from 'prop-types'
import { containerProfilePreview, profileImage, profileEmail } from './styles.css'

const ProfilePreview = ({ name, email, photoURL }) => (
  <div className={containerProfilePreview}>
    <h4>{name}</h4>
    <img src={photoURL} alt={name} className={profileImage} />
    <h5 className={profileEmail}>{email}</h5>
  </div>
)

ProfilePreview.defaultProps = {
  name: '',
  email: '',
  photoURL: '',
}

ProfilePreview.propTypes = {
  name: PropTypes.string,
  email: PropTypes.string,
  photoURL: PropTypes.string,
}

export default ProfilePreview
