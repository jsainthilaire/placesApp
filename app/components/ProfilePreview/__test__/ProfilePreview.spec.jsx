import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProfilePreview from '../ProfilePreview'

const minProps = {
  name: '',
  email: '',
  photoURL: '',
}

test('AddPlace render correctly', () => {
  const wrapper = shallow(<ProfilePreview {...minProps} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
