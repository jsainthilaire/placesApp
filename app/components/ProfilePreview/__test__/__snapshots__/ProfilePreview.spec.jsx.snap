// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlace render correctly 1`] = `
<div
  className="containerProfilePreview"
>
  <h4 />
  <img
    alt=""
    className="profileImage"
    src=""
  />
  <h5
    className="profileEmail"
  />
</div>
`;
